File: VB/WpfApplication/ViewModel.vb
Imports System
Imports System.Collections.ObjectModel
Imports DevExpress.Mvvm.POCO

Namespace WpfApplication
    Public Class ViewModel
        Public Overridable Property Items() As ObservableCollection(Of GridItem)
        Public Overridable Property CurrentItem() As GridItem

        Public Sub New()
            Items = New ObservableCollection(Of GridItem)()
            For i As Integer = 0 To 14
                Items.Add(ViewModelSource.Create(Function() New GridItem() With {.AllowEdit = i Mod 3 <> 0, .Name = String.Format("Name{0}", i), .ID = i}))
            Next i
            CurrentItem = Items(0)
        End Sub
    End Class

    Public Class GridItem
        Public Overridable Property Name() As String
        Public Overridable Property ID() As Integer
        Public Overridable Property AllowEdit() As Boolean
    End Class
End Namespace

File: CS/WpfApplication/ViewModel.cs

using System;
using System.Collections.ObjectModel;
using DevExpress.Mvvm.POCO;

namespace WpfApplication {
    public class ViewModel {
        public virtual ObservableCollection<GridItem> Items { get; set; }
        public virtual GridItem CurrentItem { get; set; }

        public ViewModel() {
            Items = new ObservableCollection<GridItem>();
            for (int i = 0; i < 15; i++)
                Items.Add(ViewModelSource.Create(() => new GridItem() { AllowEdit = i % 3 != 0, Name = String.Format("Name{0}", i), ID = i }));
            CurrentItem = Items[0];
        }
    }

    public class GridItem {
        public virtual string Name { get; set; }
        public virtual int ID { get; set; }
        public virtual bool AllowEdit { get; set; }
    }
}
